autoprecompiles/src/range_constraint_optimizer.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::BTreeMap;
+use std::fmt::Display;
+use std::hash::Hash;
+
+use num_traits::One;
+use powdr_constraint_solver::constraint_system::{
+    BusInteraction, BusInteractionHandler, ConstraintSystem,
+};
+use powdr_constraint_solver::grouped_expression::GroupedExpression;
+use powdr_constraint_solver::inliner::DegreeBound;
+use powdr_constraint_solver::range_constraint::RangeConstraint;
+use powdr_constraint_solver::solver::{new_solver, Solver};
+use powdr_number::FieldElement;
+
+pub type RangeConstraintMap<T, V> = BTreeMap<GroupedExpression<T, V>, RangeConstraint<T>>;
+
+pub trait RangeConstraintHandler<T: FieldElement> {
+    /// If the bus interaction *only* enforces range constraints, returns them
+    /// as a map of expressions to range constraints.
```

Comment:
```suggestion
    /// If the bus interaction *only* enforces range constraints, return them
    /// as a map of expressions to range constraints.
```

Reply:
Is that wrong? The function returns them, no?

Reply:
yea just thought usually function comments are written in imperative, but fine either way